dissect/target/plugins/os/windows/regf/appxdebugkeys.py

```diff
@@ -0,0 +1,122 @@
+from typing import Iterator
+
+from dissect.target.exceptions import UnsupportedPluginError
+from dissect.target.helpers.descriptor_extensions import (
+    RegistryRecordDescriptorExtension,
+    UserRecordDescriptorExtension,
+)
+from dissect.target.helpers.record import create_extended_descriptor
+from dissect.target.helpers.regutil import RegistryKey, RegistryKeyNotFoundError
+from dissect.target.plugin import Plugin, export
+
+AppxDebugKeyRecord = create_extended_descriptor([RegistryRecordDescriptorExtension, UserRecordDescriptorExtension])(
+    "windows/registry/appxdebug/key",
+    [
+        ("datetime", "ts"),
+        ("string", "name"),
+        ("string", "debug_info"),
+    ],
+)
+
+
+class AppxDebugKeysPlugin(Plugin):
+    """Plugin that iterates various AppX debug key locations."""
+
+    def _walk(self, key: RegistryKey) -> Iterator[AppxDebugKeyRecord]:
+        user = self.target.registry.get_user(key)
+        values = False
+        for value in key.values():
+            values = True
+            yield AppxDebugKeyRecord(
+                ts=key.ts,
+                name=value.name,
+                debug_info=value.value,
+                _target=self.target,
+                _key=key,
+                _user=user,
+            )
+
+        for subkey in key.subkeys():
+            values = True
+            yield from self._walk(subkey)
+
+        if not values:
+            yield AppxDebugKeyRecord(
+                ts=key.ts,
+                name=None,
+                debug_info=None,
+                _target=self.target,
+                _key=key,
+                _user=user,
+            )
```

Comment:
```suggestion
        values = key.values()
        subkeys = key.subkeys()
        
        if not values and not subkeys:
            yield AppxDebugKeyRecord(
                ts=key.ts,
                name=None,
                debug_info=None,
                _target=self.target,
                _key=key,
                _user=user,
            )
            return
        
        for value in values:
            yield AppxDebugKeyRecord(
                ts=key.ts,
                name=value.name,
                debug_info=value.value,
                _target=self.target,
                _key=key,
                _user=user,
            )

        for subkey in key.subkeys():
            yield from self._walk(subkey)
```

Why the values flag? Maybe we can "bail out" early? Or do I miss something?

Reply:
I had the assumption `key.values()` and `key.subkeys()` returned iterators, but they don't. So this is a much nicer solution indeed.